src/tree.rs

```diff
@@ -97,85 +81,48 @@ impl<H: Hasher + Default, V: Value, S: Store<V>> SparseMerkleTree<H, V, S> {
     /// Update a leaf, return new merkle root
     /// set to zero value to delete a key
     pub fn update(&mut self, key: H256, value: V) -> Result<&H256> {
-        // store the path, sparse index will ignore zero members
-        let mut path = Vec::new();
-        if !self.is_empty() {
-            let mut node = self.root;
-            loop {
-                let branch_node = self
-                    .store
-                    .get_branch(&node)?
-                    .ok_or_else(|| Error::MissingBranch(node))?;
-                let height = max(key.fork_height(branch_node.key()), branch_node.fork_height);
-                match branch_node.node_at(height) {
-                    NodeType::Pair(left, right) => {
-                        if height > branch_node.fork_height {
-                            // the merge height is higher than node, so we do not need to remove node's branch
-                            path.push((height, node));
-                            break;
-                        } else {
-                            self.store.remove_branch(&node)?;
-                            let is_right = key.get_bit(height);
-                            if is_right {
-                                node = right;
-                                path.push((height, left));
-                            } else {
-                                node = left;
-                                path.push((height, right));
-                            }
-                        }
-                    }
-                    NodeType::Single(node) => {
-                        if &key == branch_node.key() {
-                            self.store.remove_leaf(&node)?;
-                            self.store.remove_branch(&node)?;
-                        } else {
-                            path.push((height, node));
-                        }
-                        break;
-                    }
-                }
-            }
-        }
-
         // compute and store new leaf
-        let mut node = hash_leaf::<H>(&key, &value.to_h256());
+        let node = hash_leaf::<H>(&key, &value.to_h256());
```

Comment:
Because of in the new construction, we already included the node's path in the merge function, so the position of left child, right child is fixed, we can directly using leaf's value as node.